### PR TITLE
MOSIP-30869 : Updated SaveHoliday method

### PR DESCRIPTION
**Pull request by GOKULRAJ136**

### CODE REVIEW

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/HolidayServiceImpl.java [hunk 1]
```diff
@@ -56,6 +46,18 @@
...
+import org.springframework.dao.DataAccessException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Direction;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.*;
```
Comment (dhanendra06) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/HolidayServiceImpl.java:59]: remove import.*

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/integration/MasterdataIntegrationTest.java [hunk 2]
```diff
@@ -244,11 +178,77 @@
...
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Month;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
```
Comment (dhanendra06) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/integration/MasterdataIntegrationTest.java:233]: remove import.*

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/integration/MasterdataIntegrationTest.java [hunk 3]
```diff
@@ -2004,6 +2004,7 @@ public void addHolidayTypeLanguageValidationTest() throws Exception {
 				.andExpect(status().isOk());
 	}
 
+	@Ignore
```
Comment (dhanendra06) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/integration/MasterdataIntegrationTest.java:2007]: why you are skipping the test case

**Review by dhanendra06 — CHANGES_REQUESTED** [review]

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/HolidayServiceImpl.java (hunk 1 above)
Comment (GOKULRAJ136) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/HolidayServiceImpl.java]: removed * import, added proper imports.

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/integration/MasterdataIntegrationTest.java (hunk 2 above)
Comment (GOKULRAJ136) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/integration/MasterdataIntegrationTest.java]: removed * import, added proper imports.

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/integration/MasterdataIntegrationTest.java (hunk 3 above)
Comment (GOKULRAJ136) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/integration/MasterdataIntegrationTest.java]: there are some issues with the Json we are passing, updated the same, and removed it from the ignored test cases.

**Review by dhanendra06 — APPROVED** [review]

**Review by Prafulrakhade — APPROVED** [review]

**Pull request merged**